File: CS-Examples/05_Images/CompressPictures.cs
using Spire.Xls;
using System;
using System.Windows.Forms;

namespace CompressPictures
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a new workbook
            Workbook workbook = new Workbook();

            // Load an existing Excel document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\CompressPictures.xlsx");

            // Get the first worksheet in the workbook
            Worksheet sheet1 = workbook.Worksheets[0];

            // Compress the picture quality for all pictures in all worksheets
            foreach (Worksheet sheet in workbook.Worksheets)
            {
                foreach (ExcelPicture picture in sheet.Pictures)
                {
                    // Set the compression level to 50 (50% of original quality)
                    picture.Compress(50);
                }
            }

            // Specify the output file name
            string result = "CompressPictures_result.xlsx";

            // Save the modified workbook to the specified file using Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //View the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: CS-Examples/05_Images/CompressPictures.vb
Imports Spire.Xls

Namespace CompressPictures
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            'Create a new instance of the Workbook class.
            Dim workbook As New Workbook()

            'Load an Excel document from the specified file path.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\CompressPictures.xlsx")

            'Assign the first worksheet to the "sheet1" variable.
            Dim sheet1 As Worksheet = workbook.Worksheets(0)

            'Iterate through each worksheet in the workbook.
            For Each sheet As Worksheet In workbook.Worksheets
                'Iterate through each picture in the worksheet.
                For Each picture As ExcelPicture In sheet.Pictures
                    'Compresse the picture with a compression level of 50%.
                    picture.Compress(50)
                Next picture
            Next sheet

            'Specifie the file name for the resulting workbook.
            Dim result As String = "CompressPictures_result.xlsx"
            'Save the workbook to a file with the specified name and version.
            workbook.SaveToFile(result, ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
